### DISCUSSION TITLE
Use imagemagick to resize webp image is much slower than use libwebp directly

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6771] [Help]
### ImageMagick version

7.1.0-52

### Operating system, version and so on

Debian 6.3.0-18+deb9u1

### Description

I delegate libwebp for ImageMagick, but I find use ImageMagick cmd is much slower than cwebp cmd, and from the debug log, it seems Encoding part costs most of the time
![image](https://github.com/ImageMagick/ImageMagick/assets/10097167/ed1e7038-e39a-4a06-ad33-2f4264306416)

![image](https://github.com/ImageMagick/ImageMagick/assets/10097167/9a26082a-ba64-4752-85df-5b4f4330b33f)
![image](https://github.com/ImageMagick/ImageMagick/assets/10097167/926f1590-34d2-4650-b23a-0eec689b8769)

Do I need do some config for ImageMagick or libwebp?

How I install libwebp
`
tar xvzf resources/libwebp/libwebp-1.3.2.tar.gz
cd libwebp*
./autogen.sh
./configure
make
make install
ldconfig
`

 How I config ImageMagick
`
tar xvzf resources/imagick/ImageMagick-7.1.0-52.tar.gz
cd ImageMagick*
./configure --without-magick-plus-plus --without-perl --disable-openmp --with-gvc=no --with-fontconfig=yes --with-freetype=yes --with-gslib --disable-docs --with-webp=yes --with-heic=yes
make -j$(nproc)
make install
ldconfig /usr/local/lib
`

**Comment by snibgo** [comment]
IM works by decompressing the pixel data and converting it from YUV to RGB. `-resize` is high quality but expensive. Finally, pixels are converted back to YUV for the output file.

I don't know how `cwebp` works, but it may not convert from YUV to RGB and back.

Instead of `-resize`, you could `-scale` or `-sample` to improve the performance of that operation.

**Reply by ghost** [reply]
cwebp is cmd of libwebp, https://developers.google.com/speed/webp/docs/cwebp
so idealy, use cwebp cmd is just equal to IM use libwebp, but the IM is so slow. Use `-scale` or `-sample `may have different output compare with `-resize`, and I also find the Encoding of output cost most of the time by IM resize.

**Comment by snibgo** [comment]
Yes, IM's `-resize` and `-scale` and `-sample` will give different results. I don't know which is closest to cwebp resize.

Another possibility: If `magick -version` shows you have HDRI, then pixel values will be processed and stored as floating-point. A version without HDRI will process pixels with integer arithmetic, so will be faster.

**Reply by ghost** [reply]
![image](https://github.com/ImageMagick/ImageMagick/assets/10097167/1fa25569-4bf0-4b88-9858-7ac44cef86db)
so this version is with HDRI?

**Comment by snibgo** [comment]
> so this version is with HDRI?

Yes.

**Reply by ghost** [reply]
Thanks, will have a try

**Reply by ghost** [reply]
Still too slow than webp, and I have found that destory cache cost most of the process time. Is it expected?
![image](https://github.com/ImageMagick/ImageMagick/assets/10097167/47a680d9-7c12-4a71-9aa7-8316e33167d6)

**Comment by fmw42** [comment]
Please do not post images of code or error messages. They are harder to read and one cannot copy code to try to run it.

**Comment by snibgo** [comment]
>  I have found that destory cache cost most of the process time.

Why do you think that? You are logging only the cache processes, so of all the activities you are logging, a cache process will seem to take the most time.

"-debug all" might be useful.

As I explain above, I expect that cwebp will be faster than IM when processing webp files.
